extensions/cornerstone-dicom-sr/src/viewports/OHIFCornerstoneSRViewport.tsx

```diff
@@ -430,19 +437,76 @@ OHIFCornerstoneSRViewport.defaultProps = {
   customProps: {},
 };
 
+const findInstance = (measurement, displaySetService: DisplaySetService) => {
+  const {
+    displaySetInstanceUID,
+    ReferencedSOPInstanceUID: sopUid,
+  } = measurement;
+  const referencedDisplaySet = displaySetService.getDisplaySetByUID(
+    displaySetInstanceUID
+  );
+  if (!referencedDisplaySet.images) return;
+  return referencedDisplaySet.images.find(it => it.SOPInstanceUID === sopUid);
+};
+
 async function _getViewportReferencedDisplaySetData(
   displaySet,
   measurementSelected,
   displaySetService
 ) {
   const { measurements } = displaySet;
-  const measurement = measurements[measurementSelected];
 
-  const { displaySetInstanceUID } = measurement;
+  if (!displaySet.keyImageDisplaySet) {
+    const findReferences = ds => {
+      const instances = [];
+      const instanceByUrl = {};
+      for (const measurement of ds.measurements) {
+        const instance = findInstance(measurement, displaySetService);
+        if (!instance) {
+          console.log('Measurement', measurement, 'had no instances found');
+          continue;
+        }
 
-  const referencedDisplaySet = displaySetService.getDisplaySetByUID(
-    displaySetInstanceUID
-  );
+        // TODO - find actual URL
+        const { imageId } = measurement;
+        if (!imageId) continue;
+        if (instanceByUrl[imageId]) continue;
```

Comment:
shouldn't these be before you even `findInstance`? after the for right away, since we don't want to search inside all displaySets (+1000 instances) and then see oh we had it before